### PR TITLE
Embed mental imagery survey.

### LINKED ISSUE #6: Mental imagery survey is embedded in the mobile app
## Expected Behavior
When the app receives the `EVENT_STAGE_CHANGED` event and the new event stage is `MENTAL_IMAGERY`, the app should display the mental imagery survey form.

When the form is submitted, the app should return to the "thank you/waiting" screen.

## Tasks
- [ ] Event listener
- [ ] Embed form
- [ ] Waiting screen

### LINKED ISSUE #22: On the CONNECTED and EVENT_STAGE_CHANGED websocket events, the app navigates to the correct screen
## Expected Behavior
After connecting, the websocket API will send the following event.

```js
{
  event: 'CONNECTED',
  data: {
    choiceType: String, // CHOICE_COLOR | CHOICE_EMOTION_HAPPINESS | CHOICE_EMOTION_AGITATION
    choiceInverted: Boolean,
    stageId: String,
    ... // other stage-specific attributes
  },
}
```

This event contains the current stage of the event. Based on the value of `stageId`, the app should then navigate to the correct screen (if it is not there already):
* STAGE_CHOICE_IMAGERY: show the mental imagery form
* STAGE_CHOICE_COLOR_EMOTION: show the color/emotion picker
* STAGE_CHOICE_CHILLS: show the chills UI
* STAGE_END: show the results screen

`choiceType` and `choiceInverted` should be stored for later (see #25).

The app should also respond to the EVENT_STAGE_CHANGED event (same body, just different `event` value and no `choiceType` or `choiceInverted`) in the same way.

## Context
We need to be careful to do this in a way such that the user's experience is not disrupted (or even worse, data loss) if the app disconnects and reconnects during the concert.

None of these screens need to be finished as part of this story. They can be empty stubs. But the app should navigate between them.

### PR DESCRIPTION
**Pull request by cooperwalter**
## Purpose
Closes #6
Closes #22

## Approach
Qualtrics has an embedded data feature that allows hidden inputs to be populated by a query string appended to the survey URL. (See [this article](https://www.qualtrics.com/support/survey-platform/survey-module/survey-flow/standard-elements/embedded-data/).) Opening this URL in a web view allows the participant to complete the survey and for us to capture the user's ID.

Answering questions on Qualtrics doesn't change the URL. So, to listen for form submission, I configured Qualtrics to redirect the user to https://www.google.com. The WebView listens for navigation changes, and if the URL matches https://www.google.com, the app navigates the user to the thank you/waiting screen. Any URL should work, as long as it matches the URL the WebView is listening for.

**UPDATE** Using NeuroTech Foundation website instead of google

## Testing
After the app connects and receives EVENT_CONNECTED, send the following event through SNS:
```js
{
    "event": "EVENT_STAGE_CHANGED",
    "data": {
        "stageId": "STAGE_CHOICE_IMAGERY",
        "formUrl": "https://ucsf.co1.qualtrics.com/jfe/form/SV_eEBoIQ1RAp6UxdH"
    }
}
```

## Pre-Merge TODOs

- [x] ~Activate the Qualtrics survey and change the preview survey URL in the app to the final, published survey URL.~ Get formUrl from stage data
- [x] Squash that damn unhandled promise rejection from Amplify

### CODE REVIEW

**Comment by sjawhar** [comment]
I follow the changes, but it looks like we need to finish #27 first before we can be sure they will be functional. Right?

**Comment by cooperwalter** [comment]
Right! #27 should be finished before this PR.

Also, where will the uid for the user originate? Once #27 is done, I can add the uid to the redux state and set up the mental imagery web view to post it to the Qualtrics survey.

**Comment by sjawhar** [comment]
You get the user ID from the Cognito ID token:
https://github.com/sjawhar/enchanted-brain/blob/566806e3958ebb768573fd30285088410de59240/client/app/src/screens/Welcome.js#L12-L14

**Comment by cooperwalter** [comment]
If we already have the user's JWT persisted in redux state, should we skip calling Auth.currentSession() and instead immediately call concertApi.connect() with the stored JWT?

**Comment by sjawhar** [comment]
Where you store the ID token is up to you. But remember that ID tokens expire (every hour, I think), so you'll need to have a mechanism for checking that the ID token in the store isn't expired and refresh it if it is.

File: client/app/src/screens/Results.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import React, { Component } from 'react';
+import { Text, View } from 'react-native';
+import EStyleSheet from 'react-native-extended-stylesheet';
+
+class ResultsScreen extends Component {
+  render() {
+    return (
+      <View style={styles.container}>
+        <Text>I'll be the Results screen someday.</Text>
```
Comment (Borchardt) [inline client/app/src/screens/Results.js]: we're all cheering for you, little guy

**Comment by Borchardt** [comment]
Approving. Will leave it up to you whether to iterate here for unhandled promise error or track that work in another PR.

**Comment by sjawhar** [comment]
Fixed that pesky rejection. That was really bugging me.

**Pull request merged**